Library/Homebrew/extend/pathname.rb

```diff
@@ -498,6 +498,21 @@ def zipinfo
                  .encode(Encoding::UTF_8, invalid: :replace)
                  .split("\n")
   end
+
+  # Like regular `rmtree`, except it never ignores errors.
+  #
+  # This was the default behaviour in Ruby 3.1 and earlier.
+  #
+  # @api public
+  def rmtree(noop: nil, verbose: nil, secure: nil)
+    # Ideally we'd odeprecate this but probably can't given gems so let's
+    # create a RuboCop autocorrect instead soon.
+    # This is why monkeypatching is non-ideal (but right solution to get
+    # Ruby 3.3 over the line).
+    # odeprecated "rmtree", "FileUtils#rm_r"
```

Comment:
Just a comment re the autocorrect: in my experience, use of `#rmtree` in formulae typically does mean `rm_rf`, so I wonder if we could avoid the unnecessary code churn, at least for formulae.

Reply:
We are using both `rm_rf` and `rmtree` in formulae currently. Might make sense to only use `rm_rf` anyways for simplicity and explicitness.

Reply:
I'd discourage `rm_rf` unless you are ok with it not actually removing the directory.

Reply:
> in my experience, use of `#rmtree` in formulae typically does mean `rm_rf`

Not sure what you mean about this. Do we have code that makes Pathname operations ignore all errors?

Reply:
Oh right, you mean `rmtree something` rather than `something.rmtree`. Yes they mean two different things with the former meaning `rm_rf` and the latter meaning `rm_r`.

Reply:
@carlocab I think there's an argument that this could be the beginning of "monkey patch Pathname everywhere" (as we do today) to "only monkey patch Pathname and similar in formulae/casks and otherwise use things as-is".

I would really like to see us get rid of pretty much all extends of core/stdlib Ruby classes.